File: tictactoe/app/src/main/java/eu/remilapointe/tictactoe/MainActivity.java
package eu.remilapointe.tictactoe;

import android.content.DialogInterface;
import android.os.Bundle;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.view.Menu;
import android.view.MenuItem;

import static eu.remilapointe.tictactoe.R.id.board;

public class MainActivity extends AppCompatActivity {

    private BoardView boardView;
    private Board gameEngine;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(eu.remilapointe.tictactoe.R.layout.activity_main);
        boardView = (BoardView) findViewById(board);
        gameEngine = new Board();
        boardView.setGameEngine(gameEngine);
        boardView.setMainActivity(this);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(eu.remilapointe.tictactoe.R.menu.main, menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == eu.remilapointe.tictactoe.R.id.action_new_game) {
            newGame();
        }

        return super.onOptionsItemSelected(item);
    }

    public void gameEnded(char c) {
        String msg = (c == 'T') ? "Game Ended. Tie" : "GameEnded. " + c + " win";

        new AlertDialog.Builder(this).setTitle("Tic Tac Toe").
                setMessage(msg).
                setOnDismissListener(new DialogInterface.OnDismissListener() {
                    @Override
                    public void onDismiss(DialogInterface dialogInterface) {
                        newGame();
                    }
                }).show();
    }

    private void newGame() {
        gameEngine.newGame();
        boardView.invalidate();
    }


}

File: laser/app/src/main/java/com/remilapointe/laser/MainActivity.kt
package com.remilapointe.laser

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.remilapointe.laser.ui.main.achatDetailIntent
import kotlinx.android.synthetic.main.main_activity.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.main_activity)
        bt_achat.setOnClickListener{
            startActivity(achatDetailIntent())
        }

        bt_commande.setOnClickListener{
            startActivity(Intent(this, ConfigurationActivity::class.java))
        }

        bt_configuration.setOnClickListener{
            startActivity(Intent(this, ConfigurationActivity::class.java))
        }

    }
}
